File: 443. String Compression/main.py
from typing import List


class Solution:
    def compress(self, chars: List[str]) -> int:
        charsSize = len(chars)
        count = 1
        index = 0
        current = chars[0]

        for i in range(charsSize):
            if i + 1 == charsSize or chars[i] != chars[i + 1]:
                chars[index] = current
                index += 1

                if count > 1:
                    for c in str(count):
                        chars[index] = c
                        index += 1

                if i + 1 < charsSize:
                    current = chars[i + 1]
                    count = 1
            else:
                count += 1

        return index

File: 443. String Compression/main.rs
struct Solution;

impl Solution {
    pub fn compress(chars: &mut Vec<char>) -> i32 {
        let chars_size = chars.len();
        let mut count = 1;
        let mut len = 0;
        let mut current = chars[0];

        for i in 0..chars_size {
            if i + 1 == chars_size || chars[i] != chars[i + 1] {
                chars[len] = current;
                len += 1;

                if count > 1 {
                    for c in count.to_string().chars() {
                        chars[len] = c;
                        len += 1;
                    }
                }

                if i + 1 < chars_size {
                    current = chars[i + 1];
                    count = 1;
                }
            } else {
                count += 1;
            }
        }

        len as i32
    }
}
